mitodl/ocw-studio
# Create and sync sitemetadata with instructors

#### Pre-Flight checklist


#### What are the relevant tickets?
Part of #382 

#### What's this PR do?
Creates and populates a `WebsiteContent` for sitemetadata for each imported course. At the moment only instructors are read into the site metadata. The metadata for each course should have a list of text_ids which match the instructors for that course.

I updated the courses in the test data in order to have up to date versions of `instructors`. I didn't make any changes other than running the courses through ocw-data-parser and ocw-to-hugo.

#### How should this be manually tested?
 - Run `./manage.py override_site_config` or otherwise update your site config with the instructors field which was added in this PR in `ocw-course-site-config.yml`
 - Make sure you have a site called `ocw-www`. If not, create one
 - Run `./manage.py import_ocw_course_sites -b ocw-to-hugo-output-qa --limit 5`. This should take about a minute, or you can remove `--limit 5` to populate based on all the courses, which will take a half hour or so.
 - In your browser, go to one of the sites you imported, for example `1-011-project-evaluation-spring-2011`. Click on "Metadata" in the sidebar. You should see a relationfield with two instructors (see screenshot). It may take 30 seconds or so to load the instructors so you may see a uuid for a number of seconds. This should be addressed in #402 
 - You should be able to add and remove instructors, and save your changes.

#### Screenshots
![Screenshot from 2021-07-23 13-46-08](https://user-images.githubusercontent.com/863262/126821232-2ab4f4b9-fd82-48c4-ae69-574f84e4ca76.png)
